components/api/src/main/java/com/hotels/styx/api/HttpRequest.java

```diff
@@ -306,15 +370,36 @@ public InetSocketAddress clientAddress() {
 
     /**
      * Return a new {@link Builder} that will inherit properties from this request.
-     * This allows a new request to be made that will be identical to this one except for the properties
-     * overridden by the builder methods.
+     * <p>
+     * This allows a new request to be made that is identical to this one
+     * except for the properties overridden by the builder methods.
      *
      * @return new builder based on this request
      */
     public Builder newBuilder() {
         return new Builder(this);
     }
 
+    /**
+     * Aggregates content stream and converts this request to a {@link FullHttpRequest}.
+     * <p>
+     * Returns a {@link StyxObservable<FullHttpRequest>} that eventually produces a
+     * {@link FullHttpRequest}. The resulting full request object has the same
+     * request line, headers, and content as this request.
+     *
+     * The content stream is aggregated asynchronously. The stream may be connected
+     * to a network socket or some other content producer. Once aggregated, a
+     * FullHttpRequest object is emitted on the returned {@link StyxObservable}.
+     *
+     * A sole {@code maxContentBytes} argument is a backstop defence against excessively
+     * long content streams. The {@code maxContentBytes} should be set to a sensible
+     * value according to your application requirements and heap size. When the content
+     * size stream exceeds the {@code maxContentBytes}, a @{link ContentOverflowException}
```

Comment:
It would be good to transform this sentence into an `@throws ContentOverflowException` tag.

Reply:
Hmm... but it doesn't really *throw* that exception. It is just emitted thought the observable. In fact we never `toFullRequest` to throw anything except an `IllegalArgumentException`.